sync-core/src/main/java/com/cloudant/sync/query/IndexUpdater.java

```diff
@@ -0,0 +1,449 @@
+        IndexUpdater updater = new IndexUpdater(database, datastore, queue);
+
+        return updater.updateIndex(indexName, fieldNames);
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean updateAllIndexes(Map<String, Object> indexes) {
+        boolean success = true;
+
+        for (String indexName: indexes.keySet()) {
+            Map<String, Object> index = (Map<String, Object>) indexes.get(indexName);
+            ArrayList<String> fields = (ArrayList<String>) index.get("fields");
+            success = updateIndex(indexName, fields);
+            if (!success) {
+                break;
+            }
+        }
+
+        return success;
+    }
+
+    private boolean updateIndex(String indexName, ArrayList<String> fieldNames) {
+        boolean success;
+        Changes changes;
+        long lastSequence = sequenceNumberForIndex(indexName);
+
+        do {
+            changes = datastore.changes(lastSequence, 10000);
+            success = updateIndex(indexName, fieldNames, changes, lastSequence);
+            lastSequence = changes.getLastSequence();
+        } while (success && changes.size() > 0);
+
+        // raise error
+        if (!success) {
+            logger.log(Level.SEVERE, String.format("Problem updating index %s", indexName));
+        }
+
+        return success;
+    }
+
+    private boolean updateIndex(final String indexName,
+                                final ArrayList<String> fieldNames,
+                                final Changes changes,
+                                long lastSequence) {
+        if (indexName == null || indexName.isEmpty()) {
+            return false;
+        }
+
+        boolean success;
+
+        Future<Boolean> result = queue.submit( new Callable<Boolean>() {
+            @Override
+            public Boolean call() {
+                Boolean transactionSuccess = true;
+                database.beginTransaction();
+                for (BasicDocumentRevision rev: changes.getResults()) {
+                    // Delete existing values
+                    String tableName = IndexManager.INDEX_TABLE_PREFIX.concat(indexName);
+                    database.delete(tableName, " _id = ? ", new String[]{rev.getId()});
+
+                    // Insert new values if the rev isn't deleted
+                    if (!rev.isDeleted()) {
+                        // Ignoring the attachments seems reasonable right now
+                        // as we don't index them.
+                        DocumentRevisionBuilder builder = new DocumentRevisionBuilder();
+                        builder.setDocId(rev.getId());
+                        builder.setRevId(rev.getRevision());
+                        builder.setBody(rev.getBody());
+                        builder.setDeleted(rev.isDeleted());
+                        builder.setSequence(rev.getSequence());
+                        BasicDocumentRevision revision = builder.build();
```

Comment:
You don't need to do this. The iOS version does because it's mapping from a `TD_Revision` to a `CDTDocumentRevision`, but here you don't need to perform a transformation.